homeassistant/components/ambiclimate/climate.py

```diff
@@ -0,0 +1,231 @@
+                                 schema=SET_TEMPERATURE_MODE_SCHEMA)
+
+
+class AmbiclimateEntity(ClimateDevice):
+    """Representation of a Ambiclimate Thermostat device."""
+
+    def __init__(self, heater, store):
+        """Initialize the thermostat."""
+        self._heater = heater
+        self._store = store
+        self._data = {}
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._heater.device_id
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._heater.name
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            'identifiers': {
+                (DOMAIN, self.unique_id)
+            },
+            'name': self.name,
+            'manufacturer': 'Ambiclimate',
+        }
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        return self._data.get('target_temperature')
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._data.get('temperature')
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self._data.get('humidity')
+
+    @property
+    def is_on(self):
+        """Return true if heater is on."""
+        return self._data.get('power', '').lower() == 'on'
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._heater.get_min_temp()
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._heater.get_max_temp()
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def current_operation(self):
+        """Return current operation."""
+        return STATE_HEAT if self.is_on else STATE_OFF
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        await self._heater.set_target_temperature(temperature)
+
+    async def async_turn_on(self):
+        """Turn device on."""
+        await self._heater.turn_on()
+
+    async def async_turn_off(self):
+        """Turn device off."""
+        await self._heater.turn_off()
+
+    async def async_update(self):
+        """Retrieve latest state."""
+        import ambiclimate
+        try:
+            token_info = await self._heater.control.refresh_access_token()
```

Comment:
How often do we need to refresh the token?

Reply:
It will check if it is necessary and only request a new if needed: https://github.com/Danielhiversen/pyAmbiclimate/blob/master/ambiclimate/__init__.py#L343